File: feature/reference/src/main/java/in/surajsau/jisho/reference/ReferenceScreen.kt
package `in`.surajsau.jisho.reference

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import `in`.surajsau.jisho.navigation.NavigationItem
import `in`.surajsau.jisho.ui.component.AppToolbar
import `in`.surajsau.jisho.reference.components.ReferenceRow

@Composable
fun ReferenceScreen(
    modifier: Modifier = Modifier,
    onItemClicked: (NavigationItem.Resources) -> Unit,
) {
    Column(modifier = modifier) {
        AppToolbar(
            modifier = Modifier.fillMaxWidth(),
            title = "Resources"
        )

        Column(
            modifier = Modifier.fillMaxWidth()
                .padding(all = 16.dp),
            verticalArrangement = Arrangement.spacedBy(24.dp)
        ) {
            ReferenceRow(
                modifier = Modifier.fillMaxWidth(),
                color = Color(0xFFff784f),
                title = "Kana",
                description = "Hiragana & Katakana",
                onItemClicked = { onItemClicked(NavigationItem.Resources.Kana) }
            )

            ReferenceRow(
                modifier = Modifier.fillMaxWidth(),
                color = Color(0xFFdb9d47),
                title = "Kanji",
                description = "Catalog of Kanji characters",
                onItemClicked = { onItemClicked(NavigationItem.Resources.Kanji) }
            )

            ReferenceRow(
                modifier = Modifier.fillMaxWidth(),
                color = Color(0xFF3185fc),
                title = "JLPT",
                description = "Vocabulary resources for Japanese Language Proficiency Test",
                onItemClicked = { onItemClicked(NavigationItem.Resources.Jlpt) }
            )
        }
    }
}

File: iosApp/Sources/ReferenceFeature/ReferenceScreen.swift
//
//  ReferenceScreen.swift
//  
//
//  Created by surajsau on 2022/12/11.
//

import SwiftUI
import KanaFeature
import JlptFeature
import KanjiFeature
import Utils

enum Reference {
    case kana
    case kanji
    case jlpt
}

struct ReferenceScreen: View {

    @State var selectedReference: Reference?

    var body: some View {
        ZStack {
            VStack {
                VStack(spacing: 16) {
                    NavigationLink(tag: .kana, selection: self.$selectedReference) {
                        KanaResourceScreen()
                    } label: {
                        ReferenceRow(title: "Kana", description: "Hiragana & Katakana", color: Color("#FF784F"))
                    }

                    NavigationLink(tag: .kanji, selection: self.$selectedReference) {
                        KanjiResourceScreen()
                    } label: {
                        ReferenceRow(title: "Kanji", description: "Catalog of Kanji characters", color: Color("#db9d47"))
                    }
                    .isDetailLink(true)

                    NavigationLink(tag: .jlpt, selection: self.$selectedReference) {
                        JlptResourceScreen()
                    } label: {
                        ReferenceRow(title: "JLPT", description: "Vocabulary resources for Japanese Language Proficiency Test", color: Color("#3185fc"))
                    }
                    .isDetailLink(true)
                }
                .padding()

                Spacer()
            }
        }
    }
}
